nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/OAuth2Object.java

```diff
@@ -114,40 +198,171 @@ public Response getToken(@PathParam("providerId") String providerId,
         return buildResponse(Status.OK, result);
     }
 
-    private NuxeoOAuth2Token getToken(NuxeoOAuth2ServiceProvider provider, String nxuser) {
-        Map<String, Serializable> filter = new HashMap<>();
-        filter.put("serviceName", provider.getId());
-        filter.put(NuxeoOAuth2Token.KEY_NUXEO_LOGIN, nxuser);
-        return Framework.doPrivileged(() -> {
-            List<DocumentModel> entries = provider.getCredentialDataStore().query(filter);
-            if (entries != null) {
-                if (entries.size() > 1) {
-                    throw new NuxeoException("Found multiple " + provider.getId() + " accounts for " + nxuser);
-                } else if (entries.size() == 1) {
-                    return new NuxeoOAuth2Token(entries.get(0));
-                }
-            }
-            return null;
-        });
+    /**
+     * Retrieves all OAuth2 tokens for the current user.
+     *
+     * @since 9.2
+     */
+    @GET
+    @Path("token")
+    public List<NuxeoOAuth2Token> getTokens(@Context HttpServletRequest request)
+        throws IOException, RestOperationException {
+        if (!((NuxeoPrincipal)getContext().getCoreSession().getPrincipal()).isAdministrator()) {
+            return Collections.emptyList();
```

Comment:
Return `FORBIDDEN` too. No need to play nice if the user doesn't have rights to call this.